nixops/resources/vault_approle.py

```diff
@@ -0,0 +1,218 @@
+import nixops.util
+import nixops.resources
+import nixops.vault_common
+import json
+import requests
+
+
+class VaultApproleDefinition(nixops.resources.ResourceDefinition):
+    """Definition of a Vault Approle"""
+
+    @classmethod
+    def get_type(cls):
+        return "vault-approle"
+
+    @classmethod
+    def get_resource_type(cls):
+        return "vaultApprole"
+
+    def show_type(self):
+        return "{0}".format(self.get_type())
+
+
+class VaultApproleState(nixops.resources.DiffEngineResourceState):
+    """State of a Vault Approle"""
+
+    state = nixops.util.attr_property("state", nixops.resources.ResourceState.MISSING, int)
+    vault_token = nixops.util.attr_property("vaultToken", None)
+    _reserved_keys = [ "vaultAddress", "roleName", "roleId", "bindSecretId"
+                        "secretIdBoundCidrs", "tokenBoundCidrs", "policies"
+                        "secretIdNumUses", "secretIdTtl", "tokenNumUses"
+                        "tokenMaxTtl", "tokenTtl", "period", "enableLocalSecretIds"
+                        "tokenType", "secretId", "cidrList"
+                      ]
+
+    @classmethod
+    def get_type(cls):
+        return "vault-approle"
+
+    def __init__(self, depl, name, id):
+        nixops.resources.DiffEngineResourceState.__init__(self, depl, name, id)
+        self.role_name = self._state.get('roleName', None)
+        self.role_id = self._state.get('roleId', None)
+        self.vault_address = self._state.get('vaultAddress', None)
+        self.secret_id = self._state.get('secretId', None)
+
+    def show_type(self):
+        s = super(VaultApproleState, self).show_type()
+        return s
+
+    @property
+    def resource_id(self):
+        path = '/v1/auth/approle/role/'
```

Comment:
this path is being in many places, please defined once in a static variable, maybe in vault_common ?

Reply:
Yup.